[BugFix][QWEN-VL]fix wrong apply_rotary_emb_torch selection introduced by #24642

xuechendi opened the pull request:
## Purpose

#24642 introduced dispatch_rotary_emb_function method in common.py while for non_cuda or non_rocm, apply_rotary_emb_torch is differently defined in some modeling codes, so providing an argument to fix

## Test Plan

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

xuechendi committed:
Fix wrong apply_rotary_emb_torch selection

Signed-off-by: Chendi Xue <Chendi.Xue@intel.com>

ywang96 reviewed (approved):
Left some nits to reduce cognitive overhead but LGTM

xuechendi committed:
Apply change suggested

Update vllm/model_executor/layers/rotary_embedding/common.py
Update vllm/model_executor/models/qwen2_vl.py
Update vllm/model_executor/layers/rotary_embedding/common.py

Co-authored-by: Roger Wang <hey@rogerw.io>
Signed-off-by: Chendi.Xue <chendi.xue@intel.com>
Signed-off-by: Chendi Xue <Chendi.Xue@intel.com>

xuechendi committed:
Make pre-commit happy

Signed-off-by: Chendi Xue <Chendi.Xue@intel.com>

DarkLight1337 commented on vllm/model_executor/layers/rotary_embedding/common.py:
```diff
@@ -72,7 +72,9 @@ def apply_rotary_emb_dispatch(x: torch.Tensor, cos: torch.Tensor,
 @cache
-def dispatch_rotary_emb_function() -> Callable[..., torch.Tensor]:
+def dispatch_rotary_emb_function(
+    func_override: Optional[Callable[..., torch.Tensor]] = None
```
I think calling this `fallback` is more appropriate, since it's only used for other platforms

ywang96 replied:
@DarkLight1337 good point - feel free to push a naming change on this, I don't have strong opinion on it

DarkLight1337 replied:
I don't have permissions to push to the fork

ywang96 replied:
Oh - maybe @xuechendi can do it later

xuechendi replied:
But these platform are in-tree as well, why it is non-default platforms?

DarkLight1337 replied:
If it is really an override, then the provided callable should be chosen regardless of platform

xuechendi replied:
Yeah, that's right, but for some reason the PR #24642 which brought the bug did a special dispatch for apply_rotary_emb_torch for cuda and rocm.
https://github.com/vllm-project/vllm/pull/24642/files#diff-004c8b1969f7d51bc651fcc2215719914f49ca6b0fe34b4a08c00cb5fe5e5440R75-R88

As you can see, before this PR, qwen-vl is using apply_rotary_emb_torch defined inside qwen-vl, but after this PR merged, qwen-vl is using apply_rotary_emb_torch defined in common.py

DarkLight1337 replied:
Maybe call it `default` then? If no other function is given for the current platform then the default is used

xuechendi replied:
updated to `default`

xuechendi committed:
rename func_override to default

Signed-off-by: Chendi Xue <Chendi.Xue@intel.com>
